db/version_set.cc

```diff
@@ -424,129 +424,264 @@ bool SomeFileOverlapsRange(
 }
 
 namespace {
-
-// An internal iterator.  For a given version/level pair, yields
-// information about the files in the level.  For a given entry, key()
-// is the largest key that occurs in the file, and value() is an
-// 16-byte value containing the file number and file size, both
-// encoded using EncodeFixed64.
-class LevelFileNumIterator : public InternalIterator {
+class LevelIterator final : public InternalIterator {
```

Comment:
This class still feel like TwoLevelIterator with some renaming. Any chance to simplify it further?

Reply:
I tried to make TwoLevelIterator much simpler now. Maybe it can be made even simpler. Before TwoLevelIterator contains lots of special logic, like prefix may match, etc. Now the logic only goes to where it is needed.

Reply:
Sure.